compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/util/IrUtils.kt

```diff
@@ -199,8 +199,12 @@ val IrClass.primaryConstructor: IrConstructor?
 val IrDeclarationContainer.properties: Sequence<IrProperty>
     get() = declarations.asSequence().filterIsInstance<IrProperty>()
 
-val IrFunction.explicitParameters: List<IrValueParameter>
-    get() = (listOfNotNull(dispatchReceiverParameter, extensionReceiverParameter) + valueParameters)
+val IrFunction.explicitParameters: MutableList<IrValueParameter>
+    get() = mutableListOf<IrValueParameter>().apply {
```

Comment:
This change seems like something made due to performance reasons. If so I think it good if exact list size is computed before creation to avoid extra memory allocations.

Reply:
Approximation might be good enough too.